pkg/logql/evaluator.go

```diff
@@ -105,10 +114,32 @@ func GetRangeType(q Params) QueryRangeType {
 	return RangeType
 }
 
+// ParamsWithExpressionOverride overrides the query expression so that the query
+// string and the expression can differ. This is useful for for query planning
+// when plan my not match externally available logql syntax
+type ParamsWithExpressionOverride struct {
```

Comment:
This is a really nice design

Reply:
:smiling_face_with_three_hearts: I was skeptical to get a buy in here.